Windows generated node:application entry main.js file uses backslashes in import alias paths in "const manifest" - "dist" code won't run on Linux

### Current Behavior

* I have a Node CLI app that I generated with command:
  * `nx g @nrwl/node:application nodeapp1`
* I have a lib in my monorepo such as `libs/froback/`, which gets imported into `apps/nodeapp1/src/main.ts` using the import alias: `@monocod/froback`, i.e:
  * `import {echoAsciiBanner} from '@monocod/froback';`
* When running `nx serve` or `nx build` on my Windows desktop, the generated entry file `dist/apps/nodeapp1/main.js` contains code that uses `\\` backslashes as path separators:

```js
const manifest = [{ "module": "@monocod/froback", "exactMatch": "libs\\froback\\src\\index.js", "pattern": "libs/froback/src/index.ts" }, { "module": "@monocod/tsx-display", "exactMatch": "libs\\tsx-display\\src\\index.js", "pattern": "libs/tsx-display/src/index.ts" }];
```

* This runs fine on Windows, but breaks when running on Linux with error:

```
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '@monocod/froback'
Require stack:
- nodeapp1/apps/nodeapp1/src/main.js
- nodeapp1/main.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._resolveFilename (nodeapp1/main.js:32:36)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (nodeapp1/apps/nodeapp1/src/main.js:1:22)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'nodeapp1/apps/nodeapp1/src/main.js',
    'nodeapp1/main.js'
  ]
}
```

* If I manually hack the generated `dist/apps/nodeapp1/main.js` file and replace the backslashes with forward-slashes `/`, like this, then it works fine on both Windows + Linux:

```js
const manifest = [{ "module": "@monocod/froback", "exactMatch": "libs/froback/src/index.js", "pattern": "libs/froback/src/index.ts" }, { "module": "@monocod/tsx-display", "exactMatch": "libs/tsx-display/src/index.js", "pattern": "libs/tsx-display/src/index.ts" }];
```




### Expected Behavior

The generated `dist/apps/nodeapp1/main.js` file should instead always use forward-slashes so that the code works on any OS, e.g:

```js
const manifest = [{ "module": "@monocod/froback", "exactMatch": "libs/froback/src/index.js", "pattern": "libs/froback/src/index.ts" }, { "module": "@monocod/tsx-display", "exactMatch": "libs/tsx-display/src/index.js", "pattern": "libs/tsx-display/src/index.ts" }];
```

### GitHub Repo

_No response_

### Steps to Reproduce

1. Create a new app with command such as `nx g @nrwl/node:application nodeapp1`
2. Pick: `Which framework do you want to use? · none `
3. In your `apps/nodeapp1/src/main.ts` file - import something from a lib using an import alias like `@monoreponame/libname`
4. Run `nx serve` on a Windows machine
5. Try to run the code on Linux (possibly same for MacOS?): `node dist/apps/nodeapp1/main.js`

You should then see an error like:

```
Error: Cannot find module '@monoreponame/libname'
```

### Nx Report

```shell
Node   : 16.14.0
   OS     : win32 x64
   npm    : 9.6.5
   Hasher : Native

   nx (global)             : 16.0.0
   nx                      : 16.0.0
   @nx/js                  : 16.0.0
   @nrwl/js                : 16.0.0
   @nx/jest                : 16.0.0
   @nrwl/jest              : 16.0.0
   @nx/linter              : 16.0.0
   @nrwl/linter            : 16.0.0
   @nx/workspace           : 16.0.0
   @nrwl/workspace         : 16.0.0
   @nx/cypress             : 16.0.0
   @nrwl/cypress           : 16.0.0
   @nx/devkit              : 16.0.0
   @nrwl/devkit            : 16.0.0
   @nx/esbuild             : 16.0.0
   @nrwl/esbuild           : 16.0.0
   @nx/eslint-plugin       : 16.0.0
   @nrwl/eslint-plugin-nx  : 16.0.0
   @nx/express             : 16.0.0
   @nrwl/express           : 16.0.0
   @nx/node                : 16.0.0
   @nrwl/node              : 16.0.0
   @nx/react               : 16.0.0
   @nrwl/react             : 16.0.0
   @nrwl/tao               : 16.0.0
   @nx/web                 : 16.0.0
   @nrwl/web               : 16.0.0
   @nx/webpack             : 16.0.0
   @nrwl/webpack           : 16.0.0
   typescript              : 4.9.5
```


### Failure Logs

_No response_

### Additional Information

* This was occurring on Nx v15.9.2, and is still on v16.0.0 
* My guess is that perhaps `path.join()` or `path.sep` are being used for the generated code, so probably better to simply hardcode `/` in?

--- a/packages/esbuild/src/executors/esbuild/lib/build-esbuild-options.ts
+++ b/packages/esbuild/src/executors/esbuild/lib/build-esbuild-options.ts
@@ -226,7 +226,7 @@ export function getRegisterFileContent(
       // Path specifies a single entry point e.g. "a/b/src/index.ts".
       // This is the default setup.
       const { dir, name } = path.parse(pattern);
-      exactMatch = path.join(dir, `${name}${outExtension}`);
+      exactMatch = joinPathFragments(dir, `${name}${outExtension}`);
     }
     acc.push({ module: k, exactMatch, pattern });
     return acc;

